app/app.go

```diff
@@ -490,7 +491,7 @@ func NewSommelierApp(
 	// Setup an upgrade handler if doing an upgrade module upgrade
 	app.UpgradeKeeper.SetUpgradeHandler(
 		upgradeName,
-		func(ctx sdk.Context, _ upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
+		func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
 			// fromVM := make(map[string]uint64)
 			// for moduleName := range app.mm.Modules {
 			// 	fromVM[moduleName] = 1
```

Comment:
It seems like the implication in the Gaia upgrade that this code originally came from has this setting every existing module's version to 1 and then running through the full migrate path when `RunMigrations` is called. I am curious as to why that would be necessary rather than only issuing migrations from the current consensus version to any new version in the upgrade:

https://github.com/cosmos/gaia/blob/329796922bc8aedfcd5c81820357185857b23330/app/app.go#L572-L579

Every module would get run through this path:

https://github.com/cosmos/cosmos-sdk/blob/da929211d6f78014e3b40de3dc01e0ffcf0104cb/types/module/module.go#L453

which would then (since the version is set to 1) run through each migration:

https://github.com/cosmos/cosmos-sdk/blob/da929211d6f78014e3b40de3dc01e0ffcf0104cb/types/module/configurator.go#L103

and the ones deleted from the map in the Gaia example (the newly added modules) would be run through this path

https://github.com/cosmos/cosmos-sdk/blob/da929211d6f78014e3b40de3dc01e0ffcf0104cb/types/module/module.go#L458

where we are instead manually calling `InitGenesis` for the allocation module.

Reply:
Because we aren't doing a major version change of the SDK, we expect that we don't need to perform any migrations in this upgrade.

Reply:
Got it, thanks.